src/models/graph.ts

```diff
@@ -263,18 +263,41 @@ export class Graph<N extends INodeBase, E extends IEdgeBase> implements IGraph<N
     this._settings?.onMergeData?.(data);
   }
 
-  // TODO(dlozic): Add delete all mechanic.
   remove(data: Partial<{ nodeIds: number[]; edgeIds: number[] }>) {
     const nodeIds = data.nodeIds ?? [];
     const edgeIds = data.edgeIds ?? [];
 
-    this._removeNodes(nodeIds);
-    this._removeEdges(edgeIds);
+    const removedData = this._removeNodes(nodeIds);
+    const removedEdgeIds = this._removeEdges(edgeIds);
+
+    // Merge edges removed by removing nodes and by removing edges, ensuring there are no duplicate edge IDs.
+    removedData.edgeIds = removedData.edgeIds.concat(
+      removedEdgeIds.filter((edgeId) => removedData.edgeIds.indexOf(edgeId) < 0),
+    );
```

Comment:
This is the most elegant solution I've found so far, but I'm open to suggestions for any more performant/readable alternatives

Reply:
This can be slow, so I would suggest creating utility function, e.g. `dedupArrays` that receive a generic array, or multiple arrays and dedups their values - using set or map.

E.g.
```
dedupArrays([1, 2, 3], [1, 2]) -> [1, 2, 3])
dedupArrays([1, 2, 3], [1, 2], [5, 6]) -> [1, 2, 3, 5, 6])
dedupArrays([1, 2, 3], [1, 2], [5, 6], [], [6, 7]) -> [1, 2, 3, 5, 6, 7])
```